python/lsst/daf/persistence/repositoryCfg.py

```diff
@@ -0,0 +1,152 @@
+import yaml
+
+from lsst.daf.persistence import listify
+
+class RepositoryCfg(yaml.YAMLObject):
+    yaml_tag = u"!RepositoryCfg_v1"
+
+    def __init__(self, root, mapper, mapperArgs, parents, isLegacyRepository=False):
+        self._root = root
+        self._mapper = mapper
+        self._mapperArgs = mapperArgs
+        if parents is None:
+            self._parents = []
+        elif not hasattr(parents, '__iter__'):
+            self._parents = list(parents)
+        else:
+            self._parents = parents
+        self._isLegacyRepository = isLegacyRepository
+
+    @staticmethod
+    def v1Constructor(loader, node):
+        """Constructor for 'version 1' of the serlized RepositoryCfg. 
+
+        If new parameters are added to RepositoryCfg they will have to be checked for in d; if they are there
+        then their value should be used and if they are not there a default value must be used in place.
+
+        In case the structure of the serialzed file must be changed in a way that invalidates some of the
+        keys:
+        1. Increment the version number (after _v1) in the yaml_tag of this class.
+        2. Add a new constructor (similar to this one) to deserialze new serilizations of this class.
+        3. Registered the new constructor for the new version with yaml, the same way it is done at the bottom
+           of this file. 
+        4. All constructors for the older version(s) of persisted RepositoryCfg must be changed to adapt
+           the old keys to their new uses and create the current (new) version of a repository cfg, or raise a 
+           RuntimeError in the case that older versions of serialized RepositoryCfgs can not be adapted.
+        There is an example of migrating from a fictitious v0 to v1 in tests/repositoryCfg.py
+        """
+        d = loader.construct_mapping(node)
+        cfg = RepositoryCfg(root=d['_root'], mapper=d['_mapper'], mapperArgs=d['_mapperArgs'], 
+                            parents=d['_parents'], isLegacyRepository=d['_isLegacyRepository'])
+        return cfg
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return self._root == other._root and \
+               self.mapper == other._mapper and \
+               self.mapperArgs == other._mapperArgs and \
+               self.parents == other._parents and \
+               self._isLegacyRepository == other._isLegacyRepository
+
+    @property
+    def root(self):
+        return self._root
+
+    @root.setter
+    def root(self, root):        
+        if root is not None and self._root is not None:
+            raise RuntimeError("Explicity clear root (set to None) before changing the value of root.")
+        self._root = root
+
+    @property
+    def mapper(self):
+        return self._mapper
+
+    @property
+    def mapperArgs(self):
+        return self._mapperArgs
+    
+    @mapperArgs.setter
+    def mapperArgs(self, newDict):
+        self._mapperArgs = newDict
```

Comment:
There doesn't seem to be any reason to make `mapperArgs` a `property`; its getter and setter are trivial.